base/c.jl

```diff
@@ -503,3 +503,183 @@ end
 macro ccallable(rt, def)
     expand_ccallable(rt, def)
 end
+
+module CcallMacro
+
+struct NoType end
+struct CcallError{T <: AbstractString} <: Exception
+    msg::T
+end
+Base.showerror(io::IO, e::CcallError) = print(io, "CcallError: ", e.msg)
+
+getcfunc(f) = QuoteNode(f)
+getcfunc(f::Expr) = :(($(f.args[2]), $(f.args[1])))
+hashead(symbol, _) = false
+hashead(expr::Expr, head) = expr.head === head
+
+islinum(node) = node isa LineNumberNode
+
+remove_linums(e::Expr) =
+    Expr(e.head, remove_linums.(filter(!islinum, e.args))...)
+remove_linums(s) = s
+
+"""
+Determine if there are varargs and return a vector of all arguments
+based on the call signature.
+
+returns a tuple of `(arguments, varargs)`
+"""
+function getargs(call)
+    # no arguments
+    length(call.args) < 2 &&
+        return [], []
+
+    firstarg = call.args[2]
+    hashead(firstarg, :parameters) &&
+        return call.args[3:end], firstarg.args
+
+    return call.args[2:end], []
+end
+
+"""
+       mkarg(arg)
+
+`mkarg` takes an argument and returns a tuple of symbol and type. The
+argument must have a type annotation or an error will be thrown.
+"""
+function mkarg(arg)
+    !hashead(arg, :(::)) &&
+        throw(CcallError("args in @ccall need type annotations. '$arg' doesn't have one."))
```

Comment:
Use `repr(arg)` instead of `'$arg'` for clearer errors.

Seems like an `ArgumentError` would be more appropriate here.

Reply:
fixed in https://github.com/JuliaLang/julia/pull/32748/commits/bbd6c5af06e54f25126de65d9e0cae7c0d4cd846